src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/Commands/MainListPage.cs

```diff
@@ -164,6 +164,12 @@ public override void UpdateSearchText(string oldSearch, string newSearch)
                 if (_includeApps)
                 {
                     IEnumerable<IListItem> apps = AllAppsCommandProvider.Page.GetItems();
+
+                    // Remove any top level pinned apps and use the apps from AllAppsCommandProvider.Page.GetItems()
+                    _filteredItems = _filteredItems.Where(item =>
```

Comment:
Could we save some CPU cycles by just checking 

```cs
_filteredItems = _filteredItems.Where(item => item is not AppListItem);
```

having to iterate over all the apps for every item sounds long